lessons/lesson2.md

```diff
@@ -1,129 +1,65 @@
 ```
-No language indicated, so no syntax highlighting. 
-But let's throw in a <b>tag</b>.
+{
+    "name": "event-runtime-integration",
+    "description": "test event runtime",
+    "delivery_type": "WEBHOOK",
+    "webhook_url": "https://io-webhook.herokuapp.com/webhook/testjie",
+    "events_of_interest": [
+        {
+        "provider_id": "ccefc74d-9696-4b99-a799-f2d34a4189cd",
+        "event_code": "eventrt"
+        }
+    ]
+}
 ```
 
-Colons can be used to align columns.
-
-| Tables        | Are           | Cool  |
-| ------------- |:-------------:| -----:|
-| col 3 is      | right-aligned | $1600 |
-| col 2 is      | centered      |   $12 |
-| zebra stripes | are neat      |    $1 |
-
-There must be at least 3 dashes separating each header cell.
-The outer pipes (|) are optional, and you don't need to make the 
-raw Markdown line up prettily. You can also use inline Markdown.
-
-Markdown | Less | Pretty
---- | --- | ---
-*Still* | `renders` | **nicely**
-1 | 2 | 3
-
-> Blockquotes are very handy in email to emulate reply text.
-> This line is part of the same quote.
-
-Quote break.
-
-> This is a very long line that will still be quoted properly when it wraps. Oh boy let's keep writing to make sure this is long enough to actually wrap for everyone. Oh, you can *put* **Markdown** into a blockquote. 
-
-<dl>
-  <dt>Definition list</dt>
-  <dd>Is something people use sometimes.</dd>
+- After finish the steps above, you should be able to see in your terminal that you successfully create register the event, and you will also see it in at adobe developer console under the left bottom corner `event` your registration provider `eventrt` show up there. 
+![console-event](assets/console-event-3.png)
 
-  <dt>Markdown in HTML</dt>
-  <dd>Does *not* work **very** well. Use HTML <em>tags</em>.</dd>
-</dl>
 
-Three or more...
+### Event Runtime Integration 
 
----
+- With all above setup, now you get your `providerId`, `eventCode`, you can go back to your firefly App trying to invoke a custom event like below: 
+![invoke-event](assets/event-invoke-4.png)
 
-Hyphens
+- You should see this runtime action created in the `user defined actions` 
+![user-define-action](assets/user-define-action-5.png)
 
-***
+- User now adds the event api to the project to setup the event registration
+![add-event](assets/add-event-6.png)
 
-Asterisks
+- Adding from our custom event provider we just registered `eventruntime` (you should be able to see your register event in this list)
+![add-event](assets/add-event-7.png)
 
-___
+- Subscribing to the "eventrt" event type
+![add-event](assets/add-event-8.png)
 
-Underscores
+- Generate the JWT service account credentials key pair
+![add-event](assets/add-event-9.png)
 
-Here's a line for us to start with.
+- On the registration details page provide name and select the runtime user action created to setup event registration, select the user action from the dropdown of Runtime Actions 
```

Comment:
All the above steps are clear until this one...
Which action should I select here?
It should not `publish-events` because that's the publisher, not webhook consumer. So which one is the webhook handler action?

Reply:
right, fix it, should be the consume-event webhook which write in generic/index.js